### PR TITLE
Fix fastcall not applying inreg attributes to arguments

### PR DESCRIPTION
**Pull request by YaLTeR**
Fixes https://github.com/rust-lang/rust/issues/18086

### CODE REVIEW

File: src/librustc_trans/cabi_x86.rs [hunk 1]
```diff
@@ -51,4 +57,46 @@ pub fn compute_abi_info(ccx: &CrateContext, fty: &mut FnType) {
...
+
+    if flavor == Flavor::Fastcall {
+        // Mark arguments as InReg like clang does it,
+        // so our fastcall is compatible with C/C++ fastcall.
+
+        // Clang reference: lib/CodeGen/TargetInfo.cpp
+        // See X86_32ABIInfo::shouldPrimitiveUseInReg(), X86_32ABIInfo::updateFreeRegs()
+
+        // IsSoftFloatABI is only set to true on ARM platforms,
+        // which in turn can't be x86?
```
Comment (pnkfelix) [inline src/librustc_trans/cabi_x86.rs]: Seems like a reasonable deduction.

File: src/librustc_trans/cabi_x86.rs [hunk 2]
```diff
@@ -51,4 +57,46 @@ pub fn compute_abi_info(ccx: &CrateContext, fty: &mut FnType) {
...
+
+        let mut free_regs = 2;
+
+        for arg in &mut fty.args {
+            if arg.is_ignore() || arg.is_indirect() { continue; }
+
+            if arg.ty.kind() == Float {
```
Comment (eddyb) [inline src/librustc_trans/cabi_x86.rs]: Is this meant to apply to *both* `f32` and `f64` by any chance? LLVM's `float` is our `f32`.

Comment (eddyb) [inline src/librustc_trans/cabi_x86.rs]: https://github.com/llvm-mirror/clang/blob/a99e3e65489d93c72eec1870d15e1c63219f558f/lib/CodeGen/TargetInfo.cpp#L1443-L1444

So I'll have to fix this.

Comment (YaLTeR) [inline src/librustc_trans/cabi_x86.rs]: I think so, the easiest way is writing a test function which has a double as a first argument and then two more ints and comparing current Rust and clang LLVM output.

Comment (eddyb) [inline src/librustc_trans/cabi_x86.rs]: Well, I can tell you for a fact that `Float` there is `f32` - there's a `Double` variant for `f64`.

**Pull request merged**